bitcoin/src/taproot/mod.rs

```diff
@@ -1348,13 +1348,7 @@ impl fmt::Display for TaprootBuilderError {
         use TaprootBuilderError::*;
 
         match *self {
-            InvalidMerkleTreeDepth(d) => {
-                write!(
-                    f,
-                    "Merkle Tree depth({}) must be less than {}",
-                    d, TAPROOT_CONTROL_MAX_NODE_COUNT
-                )
-            }
+            InvalidMerkleTreeDepth(ref e) => write_err!(f, "invalid Merkle tree depth"; e),
             NodeNotInDfsOrder => {
                 write!(f, "add_leaf/add_hidden must be called in DFS walk order",)
```

Comment:
Unrelated but I think error messages should say what's wrong not how it should be right.

Reply:
Ah, interesting. I think all the error messages in general need work but definitely the ones in `taproot`.

Reply:
Left for another day.